rskj-core/src/main/java/co/rsk/rpc/netty/Web3HttpStatusCodeProvider.java

```diff
@@ -0,0 +1,48 @@
+package co.rsk.rpc.netty;
+
+import com.googlecode.jsonrpc4j.HttpStatusCodeProvider;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.googlecode.jsonrpc4j.ErrorResolver.JsonError;
+import static com.googlecode.jsonrpc4j.ErrorResolver.JsonError.*;
+
+public class Web3HttpStatusCodeProvider implements HttpStatusCodeProvider {
+
+    private static final List<JsonError> BAD_REQUEST_JSON_ERRORS = Arrays.asList(
+            METHOD_PARAMS_INVALID, INVALID_REQUEST, PARSE_ERROR
+    );
+
+    /**
+     * Only invalid JSON-RPC requests (e.g. a malformed JSON) qualify for a Bad Request HTTP status.
+     * In any other case, the HTTP protocol remains entirely independent from JSON-RPC (since its 2.0 version).
+     * Reference: https://www.jsonrpc.org/
+     */
+    @Override
+    public int getHttpStatusCode(int resultCode) {
+        return isBadRequestResultCode(resultCode) ?
+                HttpResponseStatus.BAD_REQUEST.code() :
+                HttpResponseStatus.OK.code();
+    }
+
+    private boolean isBadRequestResultCode(int resultCode) {
+        Optional<JsonError> optionalJsonError = BAD_REQUEST_JSON_ERRORS
```

Comment:
Why not a simple contains? Or a map of JSON ERROR to boolean (true == it is a bad request result code)?

Reply:
I didn't use a `contains` because it's a list of `JsonError` and what I receive is an integer (somehow I have to reach the `code` member variable).
Let me know what you think about the change I just uploaded (instead of using a JsonError list y directly use a collection of the codes, then I just use a `contains`).